core/src/utils/content/index.ts

```diff
@@ -0,0 +1,130 @@
+import { componentOnReady } from '../helpers';
+import { printRequiredElementError } from '../logging';
+
+const ION_CONTENT_ELEMENT_SELECTOR = 'ion-content';
+const ION_CONTENT_CLASS_SELECTOR = '.ion-content-scroll-host';
+/**
+ * Selector used for implementations reliant on `<ion-content>` for scroll event changes.
+ *
+ * Developers should use the `.ion-content-scroll-host` selector to target the element emitting
+ * scroll events. With virtual scroll implementations this will be the host element for
+ * the scroll viewport.
+ */
+const ION_CONTENT_SELECTOR = `${ION_CONTENT_ELEMENT_SELECTOR}, ${ION_CONTENT_CLASS_SELECTOR}`;
+
+const hasOverflowScroll = (node: Element | null) => {
+  if (node === null) {
+    return false;
+  }
+  const isElement = node instanceof HTMLElement;
+  const overflowY = isElement && getComputedStyle(node).overflowY;
+  // Element is scrollable if there is overflow or if the overlay is explicit
+  const isScrollable = overflowY !== 'visible' && overflowY !== 'hidden';
+  // The element is the node if it's scrollable or the scroll height is larger than the client height
+  if (isScrollable && node.scrollHeight >= node.clientHeight) {
+    return true;
+  }
+  return false;
+}
+
+const getScrollContainer = (node: Element) => {
```

Comment:
I don't get why we need to do all of this work. Since users are instructed to place `.ion-content-scroll-host` on their scrollable element, isn't querying for that class enough?

Reply:
I also have concerns over the usage of `getComputedStyle` in `hasOverflowScroll` as that will compute all of the styles for the element again.

Reply:
Agreed, I think we can cut a lot of complexity here by removing this util and requiring the implementation to correctly place the class on the scrollable element. This was to help patch "detaching ion-content", but after our team discussion, we can remove those pieces.